src/state/pportfolio/hooks.ts

```diff
@@ -0,0 +1,231 @@
+export interface Protocol {
+  id: string;
+  name: string;
+  url: string;
+  logo: string;
+}
+
+export class TokenDataUser {
+  token: Currency | Token;
+  price: number;
+  amount: number;
+  usdValue: number;
+  protocol?: Protocol;
+  logo?: string;
+
+  constructor(token: Token | Currency, price: number, amount: number, protocol?: Protocol, logo?: string) {
+    this.token = token;
+    this.price = price;
+    this.amount = amount;
+    this.usdValue = price * amount;
+    this.protocol = protocol;
+    this.logo = logo;
+  }
+}
+
+export class PairDataUser {
+  pair: Pair;
+  usdValue: number;
+  protocol?: Protocol;
+  logos?: [string, string];
+
+  constructor(pair: Pair, usdValue: number, protocol?: Protocol, logos?: [string, string]) {
+    this.pair = pair;
+    this.usdValue = usdValue;
+    this.protocol = protocol;
+    this.logos = logos;
+  }
+}
+
+// Get the USD balance of address of all chains (supported by Debank)
+export function useGetChainsBalances() {
+  const { account } = usePangolinWeb3();
+
+  const query = qs.stringify(
+    {
+      id: account,
+    },
+    {
+      encodeValuesOnly: true,
+    },
+  );
+
+  return useQuery(
+    ['getChainsBalances', account],
+    async () => {
+      if (account) {
+        const response = await openApi.get(`/total_balance?${query}`);
+        const data = response.data;
+        const chainbalances: Balances = {
+          total: data?.total_usd_value,
+          chains: [],
+        };
+        data?.chain_list?.forEach((chain: any) => {
+          const _chain = ALL_CHAINS.filter((value) => value.chain_id === chain?.community_id)[0];
+          if (_chain) {
+            chainbalances.chains.push({
+              chainID: chain?.community_id,
+              balance: chain?.usd_value,
+            });
+          }
+        });
+        return chainbalances;
+      }
+
+      return null;
+    },
+    { refetchOnWindowFocus: false },
+  );
+}
+
+// Get the Tokens of wallet
+export function useGetWalletChainTokens(chainId: number) {
+  const { account } = usePangolinWeb3();
+  let chain = getChainByNumber(chainId);
+  // This functions is temporary for Pangolin birthday
+  const getPangolinPairs = async () => {
+    const query = qs.stringify(
+      {
+        id: account,
+        protocol_id: 'avax_pangolin',
+      },
+      {
+        encodeValuesOnly: true,
+      },
+    );
+    if (account) {
+      const response = await openApi.get(`/protocol?${query}`);
+      const data = response.data;
+
+      const requestPairs: (TokenDataUser | PairDataUser)[] = data?.portfolio_item_list.map((pair: any) => {
+        const token1 = pair?.detail?.supply_token_list[0];
+        const token2 = pair?.detail?.supply_token_list[1];
+        // If token2 does not exist its because this pair is not a pair but a single staking
+        if (!token2) {
+          return new TokenDataUser(
+            new Token(
+              chainId,
+              ethers.utils.getAddress(token1?.id),
```

Comment:
lets not do getAddress here. it is automatically verifying in Token class based on the chain id passed.